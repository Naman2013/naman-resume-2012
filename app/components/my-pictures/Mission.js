import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router';
import { loadFITImages } from '../../modules/my-pictures/actions';
import styles from './Mission.scss';

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    loadFITImages,
  }, dispatch),
});

@connect(null, mapDispatchToProps)
class Mission extends Component {
  static propTypes = {
    scheduledMissionId: PropTypes.number,
    imageURL: PropTypes.string.isRequired,
    imageTitle: PropTypes.string.isRequired,
    missionDate: PropTypes.string.isRequired,
    objectIconURL: PropTypes.string.isRequired,
    fitsIsAvailable: PropTypes.bool.isRequired,
  }

  static defaultProps = {
    scheduledMissionId: -1, // set to -1 since real ID's should never be negative
  }

  handleFITSCLick = (event) => {
    event.preventDefault();
    const { scheduledMissionId } = this.props;
    this.props.actions.loadFITImages({ scheduledMissionId });
  }

  render() {
    const {
      scheduledMissionId,
      imageURL,
      imageTitle,
      missionDate,
      objectIconURL,
      fitsIsAvailable,
    } = this.props;

    return (
      <Link
        className={styles.missionContainer}
        to={`my-pictures/missions/${scheduledMissionId}`}
        style={{ backgroundImage: `url(${imageURL})` }}
      >
        <div className="content">
          <div className="row"> <b>{imageTitle} </b> <br /> {missionDate} </div>
          <div className="row"><img height="80" src={objectIconURL} alt={imageTitle} /></div>
          <div className="row">{fitsIsAvailable ? <button onClick={this.handleFITSCLick} className="fits">FITS</button> : ''}</div>
        </div>
      </Link>
    );
  }
}

export default Mission;
