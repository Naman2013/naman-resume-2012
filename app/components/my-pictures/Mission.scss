@import "../../styles/variables/fonts";
@import "../../styles/variables/colors";

.missionContainer {
  background: #000;
  height: 250px;
  width: 100%;
  margin-bottom: 30px;
  border: 4px solid transparent;
  color: $white;
  padding: 20px;
  cursor: pointer;
  opacity: 0.75;
  display: block;

  .row {
    height: 33%;
    margin: 0;

    &:first-child {
      text-align: left;
      font-size: 15px;
      font-weight: 100;
      b{
        font-size: 20px;
      }
    }

    &:nth-child(2) {
      margin: 0 auto;
      width: 80px;
      height: 80px;
    }

    &:last-child {
      display: flex;
      justify-content: flex-end;
      align-items: flex-end;

      .fits{
        color: $white;
        padding: 0 5px;
        background: $blueGreen;
        font-weight: 900;
        cursor: pointer;
      }
    }
  }

  &:hover {
    border: 4px solid $pink;
    opacity: 1;
    color: white;
  }
  &:focus {
    color: white;
    text-decoration: none;
  }
}

.MyPicturesItemRoll{
  opacity: 1;
}
