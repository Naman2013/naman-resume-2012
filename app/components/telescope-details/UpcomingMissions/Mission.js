import React from 'react';
import PropTypes from 'prop-types';
import { backgroundImageCoverMap } from '../../../styles/mixins/utilities';

function backgroundImage(imageURL) {
  return {
    ...backgroundImageCoverMap,
    backgroundImage: `url(${imageURL})`,
  };
}

const propTypes = {
  upcomingTitle: PropTypes.string.isRequired,
  upcomingObjectIconURL: PropTypes.string.isRequired,
};

const Mission = ({ upcomingTitle, upcomingObjectIconURL }) => (
  <div>
    <div className="imageIcon" style={backgroundImage(upcomingObjectIconURL)} />
    <div className="missionDetails">
      <h4 className="title">{upcomingTitle}</h4>
    </div>
  </div>
);

Mission.propTypes = propTypes;

export default Mission;
