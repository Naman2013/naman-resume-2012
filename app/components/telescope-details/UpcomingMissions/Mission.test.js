import React from 'react';
import { shallow } from 'enzyme';
import Mission from './Mission';

describe('<Mission />', () => {
  it('should render pending mission', () => {
    return;
  });

  it('should render an upcoming scheduled mission', () => {
    return;
  });
});
