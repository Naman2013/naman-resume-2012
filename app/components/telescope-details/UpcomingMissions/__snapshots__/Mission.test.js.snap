// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mission /> should render a missions title and image 1`] = `
<div>
  <div
    className="imageIcon"
    style={
      Object {
        "backgroundImage": "url(https://vega.slooh.com/icons/reservations/not_available_w.svg)",
        "backgroundPosition": "center center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "mozBackgroundSize": "cover",
        "webkitBackgroundSize": "cover",
      }
    }
  />
  <div
    className="missionDetails"
  >
    <h4
      className="title"
    >
      Andromeda Galaxy
    </h4>
  </div>
</div>
`;
